// PB2BP: BP_FILE_NAME=PbCommon
syntax = "proto3";
package idlepb;

import "defines.proto";
import "net.proto";

message Int64Pair {
	int64 v1 = 1;
	int64 v2 = 2;
}

message Int32Pair {
	int64 v1 = 1;
	int64 v2 = 2;
}

message StringInt32Pair {
	string str = 1;
	int32 value = 2;
}

message Int32Int64Pair {
	int32 v32 = 1;
	int64 v64 = 2;
}


enum ReplicatedLevelType {
	RLT_Local = 0;	// 自己所在客户端
	RLT_Offical = 1;	// 官服
	RLT_Private = 2;	// 私服
}

// 系统信息显示样式
enum SystemNoticeStyle { 
	SystemNoticeStyle_None = 0;  // 未知
	SystemNoticeStyle_Dialog = 1;  // 弹框提示
	SystemNoticeStyle_ScreenCenter = 2;  // 屏幕中央提示
}

enum SystemNoticeType {
	SNT = 0;  // 未知
	SNT_AddItem = 7;  // 添加道具
}

message PlayerData {
	int64 player_id = 1;
	string player_name = 3;
	int64 last_online_date = 4;
	int64 create_date = 5;
}

message PlayerSaveData {
	PlayerData player_data = 1;
}